File: api/src/main/java/com/ruby/rubia_server/core/dto/EnhanceTemplateDTO.java
package com.ruby.rubia_server.core.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EnhanceTemplateDTO {

    @NotNull(message = "Company ID is required")
    private UUID companyId;

    @NotBlank(message = "Original template content is required")
    private String originalContent;

    @NotBlank(message = "Enhancement type is required")
    private String enhancementType; // "friendly", "professional", "empathetic", "urgent", "motivational"

    @NotBlank(message = "Template category is required")
    private String category; // Categoria do template para contexto

    private String title; // Título do template para contexto adicional
}

File: api-dotnet/Rubia.Server/DTOs/EnhanceTemplateDto.cs
using System.ComponentModel.DataAnnotations;

namespace Rubia.Server.DTOs;

public class EnhanceTemplateDto
{
    [Required(ErrorMessage = "Company ID is required")]
    public Guid CompanyId { get; set; }

    [Required(ErrorMessage = "Original template content is required")]
    public string OriginalContent { get; set; } = string.Empty;

    [Required(ErrorMessage = "Enhancement type is required")]
    public string EnhancementType { get; set; } = string.Empty; // "friendly", "professional", "empathetic", "urgent", "motivational"

    [Required(ErrorMessage = "Template category is required")]
    public string Category { get; set; } = string.Empty; // Template category for context

    public string? Title { get; set; } // Template title for additional context
}

public class EnhancedTemplateResponseDto
{
    public string OriginalContent { get; set; } = string.Empty;
    public string EnhancedContent { get; set; } = string.Empty;
    public string EnhancementType { get; set; } = string.Empty;
    public string? AiExplanation { get; set; }
    public int TokensUsed { get; set; }
    public int CreditsConsumed { get; set; }
    public string ModelUsed { get; set; } = string.Empty;
    public DateTime EnhancedAt { get; set; } = DateTime.UtcNow;
}

public class SaveTemplateWithAiMetadataDto
{
    [Required]
    public Guid TemplateId { get; set; }

    [Required]
    [MaxLength(10000)]
    public string EnhancedContent { get; set; } = string.Empty;

    [Required]
    public string EnhancementType { get; set; } = string.Empty;

    public Guid? AiAgentId { get; set; }
    public string? AiExplanation { get; set; }
    public int? TokensUsed { get; set; }
    public int? CreditsConsumed { get; set; }
    public string? ModelUsed { get; set; }
}